src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -166,7 +164,7 @@ private class TaskLauncherActor(
     */
   private[this] def receiveDelayUpdate: Receive = {
     case RateLimiter.DelayUpdate(ref, maybeDelayUntil) if scheduledVersions.contains(ref) =>
-      val delayUntil = maybeDelayUntil.getOrElse(clock.now())
+      val delayUntil = maybeDelayUntil.map(_.deadline).getOrElse(clock.now())
```

Comment:
Maybe we should tackle this in a different update but how about

```scala
case RateLimiter.DelayUpdate(ref, Some(delay)) if scheduledVersions.contains(ref)
  val delayUntil  = delay.deadline
```

`delayUntil = clock.now()` has basically no effect. I think this is just for historical reason. @timcharper, what's your take on this?

Reply:
Tim has a different PR. Also, we may want to attach the backoff to an instance. So this should go.

Reply:
Also Tim and I were discussing yesterday and looks like new endpoint may be "overdoing" this. [See my jira comment](https://jira.mesosphere.com/browse/DCOS_OSS-4325?focusedCommentId=198862&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-198862).